### PR TITLE
CNX-9433 Remove duplication on the Connector Autofac modules again

### PR DESCRIPTION
**Pull request by adamhathcock**
CNX-9433

replacement for https://github.com/specklesystems/speckle-sharp/pull/3384 to have file moves

### CODE REVIEW

**Comment by adamhathcock** [comment]
Tested with Rhino, Revit and AutoCAD to see if I could send and receive something.  After initial fixes, things worked

File: DUI3-DX/Sdk/Speckle.Autofac/DependencyInjection/SpeckleContainerBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+    public ModuleAdapter(ISpeckleModule speckleModule)
+    {
+      _speckleModule = speckleModule;
+    }
+
+    protected override void Load(ContainerBuilder builder) => _speckleModule.Load(CreateInstanceForLoading(builder));
+  }
+
+  public SpeckleContainerBuilder AddModule(ISpeckleModule module)
```
Comment (JR-Morgan) [inline DUI3-DX/Sdk/Speckle.Autofac/DependencyInjection/SpeckleContainerBuilder.cs:104]: What was the purpose of creating all these `AddModule`, `AddSingleton`, `AddTransient` wrappers?
Why not simply have callers call `ContainerBuilder.RegisterInstance`? 

Seems like we'll have to duplicate a lot of autofac functions and I'm not sure I understand the purpose of this?

File: DUI3-DX/Sdk/Speckle.Autofac/EnumerationExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Speckle.Autofac;
+
+public static class EnumerationExtensions
+{
+  public static void ForEach<T>(this IEnumerable<T> items, Action<T> action)
```
Comment (JR-Morgan) [inline DUI3-DX/Sdk/Speckle.Autofac/EnumerationExtensions.cs:5]: Doesn't linq already have effectively this?
https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.foreach?view=netstandard-2.0

File: DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/HostApp/RhinoDocumentStore.cs [hunk 3]
```diff
@@ -9,8 +9,8 @@ public class RhinoDocumentStore : DocumentModelStore
   private const string SPECKLE_KEY = "Speckle_DUI3";
   public override bool IsDocumentInit { get; set; } = true; // Note: because of rhino implementation details regarding expiry checking of sender cards.
 
-  public RhinoDocumentStore(JsonSerializerSettings jsonSerializerSettings, bool writeToFileOnChange)
-    : base(jsonSerializerSettings, writeToFileOnChange)
+  public RhinoDocumentStore(JsonSerializerSettings jsonSerializerSettings)
+    : base(jsonSerializerSettings, true)
```
Comment (JR-Morgan) [inline DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/HostApp/RhinoDocumentStore.cs:13]: What was the original intention with this bool? @AlanRynne

File: DUI3-DX/Sdk/Speckle.Autofac/EnumerationExtensions.cs (hunk 2 above)
Comment (JR-Morgan) [reply DUI3-DX/Sdk/Speckle.Autofac/EnumerationExtensions.cs:5]: Ah, its not linq sorry, just `List<T>`
Out of interest, do you really prefer this over just using a regular `ForEach` loop? is this not just slower (due to closure captures)?

**Review by JR-Morgan — APPROVED** [review]

**Pull request merged**